doc/sphinx/source/esmvalcore/preprocessor.inc

```diff
@@ -1,36 +1,218 @@
+ozone as observed variable (toz), but CMIP models only provide the ozone 3D
+field. In this case, a derivation function is provided to vertically
+integrate the ozone and obtain total column ozone for direct comparison with
+the observations.
 
-Time manipulation
-=================
-The _time.py module contains the following preprocessor functions:
+The user contributing a new derived variable also needs to define
+a name for it and to provide the corresponding CMOR table. This is to guarantee
+the proper metadata definition is attached to the derived data. Such custom
+CMOR tables are collected as part of the ESMValTool core package. By default, 
+the variable derivation will be applied only if not already available in the
+input data, but the derivation can be forced by setting the appropriate flag.
+
+.. code-block:: yaml
+
+  variables:
+    toz:
+      derive: true
+      force_derivation: false
+
+The required arguments for this module are two boolean switches:
+* derive: activate variable derivation
+* force_derivation: force variable derivation even if the variable is
+directly available in the input data.
+
+.. warning::
+   Link to function missing.
+
+
+
+CMOR check and dataset-specific fixes
+======================================
+.. warning::
+   Documentation of _reformat.py, check.py and fix.py to be added
+
+
+Vertical interpolation
+======================
+.. warning::
+   Documentation of  _regrid.py (part 1) to be added
+
+
+Land/Sea/Ice Masking
+====================
+
+Certain metrics and diagnostics need to be computed and performed on specific
+domains on the globe. The ESMValTool preprocessor supports filtering
+the input data on continents, oceans/seas and ice. This is achived by masking
+the model data and keeping only the values associated with grid points that
+correspond to, e.g., land, ocean or ice surfaces, as specified by the
+user. Where possible, the masking is realized using the standard mask files
+provided together with the model data as part of the CMIP data request (the
+so-called fx variable). In the absence of these files, the Natural Earth masks
+are used: although these are not model-specific, they represent a good
+approximation since they have a much higher resolution than most of the models
+and they are regularly updated with changing geographical features.
+
+In ESMValTool, land-sea-ice masking can be done in two places: in the
+preprocessor, to apply a mask on the data before any subsequent preprocessing
+step and before running the diagnostic, or in the diagnostic scripts
+themselves. We present both these implementations below.
+
+To mask out a certain domain (e.g., sea) in the preprocessor,
+`mask_landsea` can be used:
+
+.. code-block:: yaml
+
+    preprocessors:
+      preproc_mask:
+        mask_landsea:
+          mask_out: sea
+
+and requires only one argument:
+* mask_out: either land or sea.
+
+The preprocessor automatically retrieves the corresponding mask (`fx: stfof` in
```

Comment:
I think this is not entirely accurate, in some cases the other fx file may be used. Maybe @schlunma can explain, because he recently changed this code?

Reply:
I think this is correct. I can't remember changing anything in here, I rather changed the use of fx files in the `area_statistics` preprocessor and in the derivation of some variables.